pkg/remote/client_test.go

```diff
@@ -172,6 +175,40 @@ func TestNewWriteRequestBody(t *testing.T) {
 	assert.Contains(t, string(b), `label1`)
 }
 
+func TestNewWriteRequestBodyWithStaleMarker(t *testing.T) {
+	t.Parallel()
+
+	ts := []*prompb.TimeSeries{
+		{
+			Labels: []*prompb.Label{{Name: "label1", Value: "val1"}},
+			Samples: []*prompb.Sample{{
+				Value:     math.Float64frombits(0x7ff0000000000002),
+				Timestamp: time.Now().UnixMilli(),
```

Comment:
Maybe store this timestamp or use a fixed value, so that you can assert you got the same value back below, instead of the current `NotEmpty()` assertion?